packages/main/src/plugin/util/manifest.ts

```diff
@@ -0,0 +1,47 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { ImageInfo } from '../api/image-info.js';
+
+const GUESSED_MANIFEST_SIZE = 1000000; // 1MB
+
+// Function to safely "guess" if the image is actually a manifest
```

Comment:
do you think a manifest could reach that 1MB limit ? looks like it's much more less than that

Reply:
Probably not, maybe 50KB max?